gax/src/main/java/com/google/api/gax/retrying/StreamingRetryAlgorithm.java

```diff
@@ -76,6 +97,37 @@ public TimedAttemptSettings createNextAttempt(
     return super.createNextAttempt(prevThrowable, prevResponse, prevSettings);
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>The attempt settings will be reset if the stream attempt produced any messages.
+   */
+  @Override
+  public TimedAttemptSettings createNextAttempt(
```

Comment:
Can we make these methods (context-aware and context-unaware methods in streaming algorithm) call each other, like it is done in the non-streaming versions? This is to reduce code dupliation and make the implementation more robust in general.

Reply:
Done (this is now possible as the context-aware methods accept null values for the context)